otsclient/cmds.py

```diff
@@ -173,22 +193,23 @@ def stamp_command(args):
         # Remember that the files - and their timestamps - might get separated
         # later, so if we didn't use a nonce for every file, the timestamp
         # would leak information on the digests of adjacent files.
-        nonce_appended_stamp = file_timestamp.timestamp.ops.add(OpAppend(os.urandom(16)))
+        nonce = bytes.fromhex(args.nonce) if args.nonce else os.urandom(16)
+        nonce_appended_stamp = file_timestamp.timestamp.ops.add(OpAppend(nonce))
```

Comment:
IIUC if args.nonce is used without txid, this will lead to a corrupt OTS file. Also, IIUC, txid can be used without nonce. There should be an explicit test to make sure that they're both used at the same time, or not at all.

Reply:
There's some python magic that can make certain combinations of arguments mandatory. I'll see if I can figure it out, but clues are welcome.

Reply:
I think it's fine to just check for that condition and quit with an error. We don't absolutely need to do it in argument handling.

Reply:
Added a check